### ISSUE TITLE
[Data sync status] Improve sync status labels

### ISSUE THREAD

**Issue by gino-m** [google/ground-android#2866]
@jo-spek @jabramowitz5 We spoke about this but I can't find an FR or notes.

| Before    | After |
| -------- | ------- |
| Upload pending | Pending |
| Failed | Pending retry |
| Syncing | Uploading form data |
| Data synced\n(media pending) | Photos pending |
| Data synced\n(media pending retry) | Photos pending retry |
| Media syncing | Uploading photos |
| Synced | Upload done |

We should also precede the site name with the marker icon and move it above the user's name data, since it's the primary entity to which the submissions are attached.

Current screen:

![Image](https://github.com/user-attachments/assets/0969b113-c6aa-4a76-bb49-6d09a80b3e74)

**Comment by amegantz** [comment]
1. "Uploading form data" can just be "Uploading data"
2. Are other media types (e.g. video) supported? If so, I'd keep "media" as is. Otherwise, everything else LGTM

**Comment by rawbzz** [comment]
I think i remember @gino-m you talking about grouping these also by survey? I took a stab at rearranging some of the type hierarchy![Image](https://github.com/user-attachments/assets/1e0d9a29-19ae-491f-bf35-463cf1d8fb8e)

**Comment by jo-spek** [comment]
> [@jo-spek](https://github.com/jo-spek) [@jabramowitz5](https://github.com/jabramowitz5) We spoke about this but I can't find an FR or notes.

Do you mean the survey name? --> Still pertains to issue https://github.com/google/ground-android/issues/2690. Without a textblock feature, giving text instructions remains tricky.

If you mean creating an FR for changing the terms (Synced --> Uploaded, etc...), I may just quickly change the strings.xml and do a pull request.

**Comment by gino-m** [comment]
> > [@jo-spek](https://github.com/jo-spek) [@jabramowitz5](https://github.com/jabramowitz5) We spoke about this but I can't find an FR or notes.
> 
> Do you mean the survey name? --> Still pertains to issue [#2690](https://github.com/google/ground-android/issues/2690). Without a textblock feature, giving text instructions remains tricky.
> 
> If you mean creating an FR for changing the terms (Synced --> Uploaded, etc...), I may just quickly change the strings.xml and do a pull request.

Yes, I was referring to differentiating upload tasks by survey name/title. I think this issue is sufficient to track. Not sure how this is related to #2690.

**Comment by jo-spek** [comment]
Ah, I only see it now. Was too stuck on the awfully long title in picture 1. Ok!

**Comment by gino-m** [comment]
> 1. "Uploading form data" can just be "Uploading data"

@amegantz  In that case we might as well just say "Uploading", since I don't think adding "data" will help the user understand it's only the form data that's uploading

> 2. Are other media types (e.g. video) supported? If so, I'd keep "media" as is. Otherwise, everything else LGTM

Only photos for now.

**Comment by gino-m** [comment]
@amegantz And I spoke on chat yesterday; we probably need to take a step back and simplify the experience. A great example of how this could work from Drive (crroping out UI elements which don't apply here):

When there are uploads but no connection, a message is shown:
![Image](https://github.com/user-attachments/assets/803427e7-07e9-4b4d-abf6-5abada9f4b41)

An uploads appear as:

![Image](https://github.com/user-attachments/assets/b37bd9b3-c45f-4905-978e-2a197ba4b06e)

Uploading:

![Image](https://github.com/user-attachments/assets/b74d9bce-f40d-4ca7-b4ab-676ea2215f76)

We don't have a thumbnail image for uploads, but could show a spinner on the right-hand side. We could initially use an infinite spinner, then replace with an actual progress spinner.

When done:

![Image](https://github.com/user-attachments/assets/f7fbd504-3bbc-4460-955c-2dd91407b9f9)

Note the upload time, which we currently don't have.

**Comment by gino-m** [comment]
@kenstershiro This is an incremental improvement, not sure it needs full pm/ux review. Please add back the label if you disagree.

**Comment by rawbzz** [comment]
![Image](https://github.com/user-attachments/assets/c93b5e52-b98a-465b-970b-703de144fd38)

**Comment by gino-m** [comment]
Nice work, @rawbzz! A few suggestions:

* Can we remove the upload status label on the right?
* I assume "data type" is a placeholder for job title and site/LOI name/ID. How do we show those two bits of information?
* Should "Uploaded" date be in its own column? (Maybe the status column?)
*

**Comment by gino-m** [comment]
> If uploaded date is its own column, where would the status's go (synced /uploading) or do we not need that text and just have icons?

I think just having icons is sufficient. For the layout w/o columns, perhaps the essential info can be stacked, similar to how it's shown in the Google Drive upload screen? (partial snippets above)

**Comment by rawbzz** [comment]
like this?
<img width="351" alt="Image" src="https://github.com/user-attachments/assets/1cd0fd94-1baf-41c7-9906-12d3c0386d8d" />

**Comment by gino-m** [comment]
Yes! That feels simpler and cleaner IMO. Should we still show both "Collected/saved" date in addition to "Uploaded" date?

Would "Job name" and "Site ID/name" show in the place of "Date type"..? On one line or two? 

Example job: "Map agricultural areas"
Example site ID/name: "IDX0001"

**Comment by rawbzz** [comment]
<img width="303" alt="Image" src="https://github.com/user-attachments/assets/7386d703-c8fd-44e5-aef9-576d480b5186" />

**Comment by gino-m** [comment]
Thank you, @rawbzz, this is great! When you get a chance could we also see a variant without the additional line spacing around the middle and outer blocks of text?

**Comment by rawbzz** [comment]
Yup!
<img width="196" alt="Image" src="https://github.com/user-attachments/assets/91d2be33-f401-4247-8c4c-7ece065f7320" />

**Comment by gino-m** [comment]
> Yup! <img alt="Image" width="196" src="https://private-user-images.githubusercontent.com/108412767/397473520-91d2be33-f401-4247-8c4c-7ece065f7320.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ2MzAyODAsIm5iZiI6MTczNDYyOTk4MCwicGF0aCI6Ii8xMDg0MTI3NjcvMzk3NDczNTIwLTkxZDJiZTMzLWY0MDEtNDI0Ny04YzRjLTdlY2UwNjVmNzMyMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIxOVQxNzM5NDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kMjY2ZGRjZDUxMDYxZTI5N2I1ODkzMjUzMTM4MThhODNmNTgwMTQ5MjNjYTBhOWJjMDMzOTVhNDEyOGVkMjg0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.OrrX365oj4rezpEijiup-cHWJjoeDvoF-ZYWYNOrIX0">

Thanks! This looks more intentional to me, but a bit crowded. Is the line spacing the same on all lines? Can we increase it slightly?

**Comment by rawbzz** [comment]
<img width="351" alt="Image" src="https://github.com/user-attachments/assets/2f4a3db0-c228-4af1-8848-cdcc657e5441" />

**Comment by jabramowitz5** [comment]
@rawbzz This looks great! Would the text break onto multiple lines for long job names (for example the job name @gino-m used in the initial mock up, which is our current job name for Ghana Lacuna Fund project)?
